core/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -50,6 +50,7 @@
     private String currentNodeId;
     private String relocatingNodeId;
     private boolean primary;
+    private long primaryTerm;
```

Comment:
OK. I gave it some thought and I agree it will be simpler, at least for now (may change when we start using the term signaling primary replica resync) to have it out of the shard routing. Biting the bullet and changing this...